src/pages/workspaces/workspace/notebooks/NotebookLauncher.js

```diff
@@ -0,0 +1,150 @@
+import _ from 'lodash/fp'
+import { Fragment } from 'react'
+import { div, h } from 'react-hyperscript-helpers'
+import { icon, spinner } from 'src/components/icons'
+import { TopBar } from 'src/components/TopBar'
+import { Rawls, Leo } from 'src/libs/ajax'
+import { getBasicProfile } from 'src/libs/auth'
+import { reportError } from 'src/libs/error'
+import * as Style from 'src/libs/style'
+import * as Nav from 'src/libs/nav'
+import * as Utils from 'src/libs/utils'
+import { Component } from 'src/libs/wrapped-components'
+
+
+const wait = ms => new Promise(resolve => setTimeout(resolve, ms))
+
+
+const styles = {
+  pageContainer: {
+    padding: '2rem'
+  },
+  step: {
+    container: {
+      display: 'flex',
+      alignItems: 'center',
+      lineHeight: '2rem',
+      margin: '0.5rem 0'
+    },
+    col1: {
+      flex: '0 0 30px'
+    },
+    col2: {
+      flex: 1
+    }
+  }
+}
+
+
+class NotebookLauncher extends Component {
+  constructor(props) {
+    super(props)
+
+    this.creator = getBasicProfile().getEmail()
+  }
+
+  async componentDidMount() {
+    const bucketName = await this.resolveBucketName()
```

Comment:
In any case, this logic should be wrapped in a `try` so we can stop it if one of the steps fails.